src/main/java/com/onwelo/tinvalidator/validator/EEValidator.java

```diff
@@ -0,0 +1,44 @@
+package com.onwelo.tinvalidator.validator;
+
+import java.util.regex.Pattern;
+
+public class EEValidator implements Validator {
+
+    private static final Pattern PATTERN = Pattern.compile("[1-6]\\d{2}[01]\\d[0-3]\\d[0-7]\\d{3}");
+
+    public Pattern getPattern(){
+        return PATTERN;
+    }
+
+    public boolean computeControlSum(String tin) {
+        int[] weights = {1, 2, 3, 4, 5, 6, 7, 8, 9, 1};
+        try {
+            assert Integer.parseInt(tin.substring(3,5)) <= 12;
+            assert Integer.parseInt(tin.substring(3,5)) > 0;
+            assert Integer.parseInt(tin.substring(5,7)) <= 31;
+            assert Integer.parseInt(tin.substring(5,7)) > 0;
+            assert Integer.parseInt(tin.substring(7,10)) <= 710;
+            assert Integer.parseInt(tin.substring(7,10)) > 0;
+
+            int sum = 0;
+            for (int i = 0; i < weights.length; i++) {
+                sum += Integer.parseInt(tin.substring(i, i + 1)) * weights[i];
+            }
+            if (sum % 11 != 10) {
```

Comment:
"If remainder is less than 10, the remainder is the check digit;" - why you check if remainder is not equal instead of if it's less then?

Reply:
In this situation, right after modulo 11 it is impossible to get value other than exactly 10, which is not lesser than 10, thus `sum % 11 != 10` and `sum % 11 < 10` notations are equivalent here.